### PR TITLE
[Applab Data] Synchronous versions of data blocks

### PR DESCRIPTION
**Pull request by ajpal**
Synchronous versions of data blocks
These four blocks already exist and are synchronous:
image
I had never seen the `startWebRequestSync` block- looks like it was added back in 2018 ([link](https://github.com/code-dot-org/code-dot-org/pull/26191)), not sure if it was ever used. It wasn't working when I added it to a level, but it was pretty simple to fix (first commit).

The blocks are new:
image

The functionality is exactly the same as the async versions, it just uses the `nativeIsAsync` property to make the call "blocking"

## Links

[jira](https://codedotorg.atlassian.net/browse/STAR-1355)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/applab/api.js [hunk 1]
```diff
@@ -328,8 +328,8 @@ export function startWebRequest(url, func) {
   return Applab.executeCmd(null, 'startWebRequest', {url: url, func: func});
 }
 
-export function startWebRequestSync(url, headers, func) {
```
Comment (maddiedierker) [inline apps/src/applab/api.js:331]: it looks like this param was added in #26383 to allow levelbuilders/users to make requests that require headers. i'm not sure if this is still needed, but is there a way for us to double-check? (i honestly don't know the answer to that, but we may be able to ask curriculum?)

File: apps/src/applab/api.js [hunk 2]
```diff
@@ -328,8 +328,8 @@ export function startWebRequest(url, func) {
...
-  return Applab.executeCmd(null, 'startWebRequestSync', {url, func, headers});
+export function startWebRequestSync(url, func) {
+  return Applab.executeCmd(null, 'startWebRequestSync', {url, func});
```
Comment (maddiedierker) [inline apps/src/applab/api.js]: the other new synchronous commands execute the async function but use the `nativeIsAsync` dropletConfig property to make them blocking -- could we do the same here now that the sync/async command signatures are the same? (i.e., execute `startWebRequest` here and remove [the `startWebRequestSync` command from applab/commands.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/applab/commands.js#L1707))

File: apps/src/applab/api.js [hunk 3]
```diff
@@ -328,8 +328,13 @@ export function startWebRequest(url, func) {
   return Applab.executeCmd(null, 'startWebRequest', {url: url, func: func});
 }
 
-export function startWebRequestSync(url, headers, func) {
-  return Applab.executeCmd(null, 'startWebRequestSync', {url, func, headers});
+export function startWebRequestSync(url, func) {
+  return Applab.executeCmd(null, 'startWebRequest', {
+    url: url,
+    func: (status, contentType, responseText) => {
+      func(responseText);
```
Comment (maddiedierker) [inline apps/src/applab/api.js:335]: just making sure i understand -- this means that in a user's project, the code would look like...

```javascript
var myResponseText = startWebRequestSync(myUrl);
```

...right?

**Review by maddiedierker — APPROVED** [review]

File: apps/src/applab/api.js (hunk 3 above)
Comment (ajpal) [reply apps/src/applab/api.js:335]: Yeah, that's correct

**Pull request merged**